Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ccl/dccwalletinfo/update/DccWalletInfoUpdateTrigger.kt

```diff
@@ -2,16 +2,41 @@ package de.rki.coronawarnapp.ccl.dccwalletinfo.update
 
 import de.rki.coronawarnapp.ccl.dccwalletinfo.update.DccWalletInfoUpdateTask.DccWalletInfoUpdateTriggerType.TriggeredAfterCertificateChange
 import de.rki.coronawarnapp.ccl.dccwalletinfo.update.DccWalletInfoUpdateTask.DccWalletInfoUpdateTriggerType.TriggeredAfterConfigUpdate
+import de.rki.coronawarnapp.covidcertificate.common.certificate.CertificateProvider
 import de.rki.coronawarnapp.tag
 import de.rki.coronawarnapp.task.TaskController
 import de.rki.coronawarnapp.task.common.DefaultTaskRequest
+import de.rki.coronawarnapp.util.coroutine.AppScope
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.plus
 import timber.log.Timber
 import javax.inject.Inject
+import javax.inject.Singleton
 
+@Singleton
 class DccWalletInfoUpdateTrigger @Inject constructor(
-    private val taskController: TaskController
+    private val taskController: TaskController,
+    certificateProvider: CertificateProvider,
+    @AppScope appScope: CoroutineScope,
+    dispatcherProvider: DispatcherProvider
 ) {
 
+    init {
+        certificateProvider.certificateContainer
```

Comment:
I don't think that:
1- This is the right provider it should `PersonCertificatesProvider` , not `CertificatesProvider` we should calculate the DccWalletInfo after grouping the certificates. Relying on the certificates repos to trigger calculation is not right unless we do the re-grouping here too without waiting for `PersonCertificatesProvider` to do that 

2- You do trigger the calculation on `EVERY` change in the flow (or latest since you are using `conflate` ) ,but that is still not right , calculation should be triggered on changes for the following events:
   - Register a Dcc
   - Restore a Dcc
   - Recycle a Dcc
   - Retrieve a Dcc
   
   and in `2.19` after new re-grouping

Reply:
is the _PersonCertificatesProvider_ doing the regrouping in 2.19 ?

Reply:
Yes